content/docs/developers/_index.md

```diff
@@ -0,0 +1,145 @@
+user to a specific room when it receives a room join event). The expected
+behaviour of homeservers is described fully in the [Matrix Specification](https://spec.matrix.org).
+
+You can find a list of existing homeserver implementations in the
+[Ecosystem > Servers](/ecosystem/servers) section of this website. Some of them
+are open source, so you can explore how they work. Please also refer to their
+documentation if you want to deploy them either for testing or in production.
+
+### Client
+
+Homeservers communicate between each other with the Server-Server / Federation
+API, but they also communicate with clients in a standard way: the Client-Server
+API.
+
+Clients are pieces of software who can use a Matrix account to send and receive
+events from a specific homeserver. The clients themselves only ever talk to the
+homeserver of the account they're using. If a client uses the
+`@alice:example.com` account, they will only talk to `example.com`.
+
+The most common kinds of client are the user facing ones. In the case of instant
+messaging, those clients show rooms as timeline of messages, with users joining,
+leaving, redacting messages…
+
+To get a better idea of what clients look like in practice, you can find a list
+on [Ecosystem > Clients](/ecosystem/clients) and give them a go.
+
+If you're more interested in writing your own client to bring a new experience
+to users, you may want to rely on an existing SDK (see
+[Ecosystem > SDKs](/ecosystem/sdks)). Those will do significant part of the
+Matrix heavy-lifting and allow you to focus on the UX you want to build.
+Finally, if you're interested in learning more about the interactions between
+clients and servers, please head to
+[the Client-Server section of the Matrix Specification](https://spec.matrix.org/latest/client-server-api/).
+
+### AppService (bots and bridges)
+
+Many Matrix bots are non-human clients. They can be built with the same SDKs as
+regular clients, and instead of showing a UI to display what is happening they
+will listen to events, parse them, and for example send automatic replies.
+
+#### Simple and advanced bots
+
+A good example of simple bot would be a RSS bot: it subscribes to a RSS feed
+completely outside of Matrix and whenever it sees a new item in the feed it
+posts a message in a specific Matrix room, with the name of the item. Such a bot
+is nothing more than a very limited and specialised client.
+
+But sometimes you need to get a more global view of what is happening on your
+homeserver to take action. If you want to write an anti-spam module for example,
+you want to be able to read each and every message from public rooms to detect
+patterns and ring the alarm or take action directly.
+
+To do it with a bot, you would need to invite the bot in each and every room
+where you want the monitoring to happen. An appservice is able to monitor all
+the _unencrypted_ events (messages sent/edited/redacted, people joining or 
+leaving rooms).
```

Comment:
It is? 😄

Don't you need either the main user or a user from the appservice's user namespace in the room?

Reply:
From the bridge crew:

>  For your appservice to monitor what's happening in a room, its HS needs to have a local copy of the room, so a local user needs to be said room, you don't need to have a user from within the namespace in the room

> if my appservice has the namespace foo-*, it will be able to monitor all the unencrypted events within all the rooms starting with #foo- that my HS has a local copy of, even if there is no @foo-*:mydomain.tld user in them

But you're right I need to say an appservice is bound to their namespace 👍